auth0/v3/authentication/token_verifier.py

```diff
@@ -340,3 +339,135 @@ def _verify_payload(self, payload, nonce=None, max_age=None, organization=None):
                     'Authentication Time (auth_time) claim in the ID token indicates that too much '
                     'time has passed since the last end-user authentication. Current time ({}) '
                     'is after last auth at ({})'.format(now, auth_valid_until))
+
+
+class AccessTokenVerifier():
+    """Class that verifies JWT access tokens following the steps defined in the
+    RFC 7519, for more details see:
+    https://datatracker.ietf.org/doc/html/rfc7519#section-7.2
+
+    Args:
+        signature_verifier (SignatureVerifier): The instance that knows how to
+        verify the signature.
+        issuer (str): The expected issuer claim value.
+        audience (str): The expected audience claim value.
+        leeway (int, optional): The clock skew to accept when verifying date
+        related claims in seconds. Defaults to 60 seconds.
+    """
+
+    def __init__(self, signature_verifier, issuer, audience, leeway=0):
+        instance_check = isinstance(signature_verifier, SignatureVerifier)
+        if not signature_verifier or not instance_check:
+            raise TypeError("signature_verifier must be an instance of "
+                            "SignatureVerifier.")
+
+        self.iss = issuer
+        self.aud = audience
+        self.leeway = leeway
+        self._sv = signature_verifier
+        self._clock = None  # visible for testing
+
+    def verify(self, token, scopes=None, permissions=None):
+        """Attempts to verify the given JWT access token, following the steps
+        defined in the RFC 7519, for more details see:
+        https://datatracker.ietf.org/doc/html/rfc7519#section-7.2
+        Auth0 only uses JWT access tokens.
+
+        Args:
+            token (str): The JWT access token to verify.
+            scopes (str, optional): A string of scopes to verify.
```

Comment:
I find it a bit strange that this one is named "scopes" (plural) and the type is a string, versus "permissions" (also plural) where the type is an array of strings. I'd try to be consistent about it. And also, if we keep the "string" types like in scope, clarify how we expect them to be separated (using whitespace)